src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncDispatcher.cs

```diff
@@ -82,12 +74,13 @@ public async Task Start(CancellationToken cancellationToken)
                             if (Logger.IsWarn) Logger.Warn("Dormant task is NULL when trying to await it");
                         }
 
-                        await (dormantTaskLocal?.Task ?? Task.CompletedTask);
+                        await (dormantTaskLocal?.Task ?? Task.CompletedTask).WaitAsync(cancellationToken);
                         if (Logger.IsDebug) Logger.Debug($"{GetType().Name} got activated.");
                     }
                     else if (currentStateLocal == SyncFeedState.Active)
                     {
-                        T request = await (Feed.PrepareRequest() ?? Task.FromResult<T>(default!)); // just to avoid null refs
+                        cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Same here

Reply:
Caught below, my intention was to make everything based on cancellation of the token, so the code may choose the proper moment of interruption